modin/pandas/series_utils.py

```diff
@@ -296,6 +298,18 @@ def match(self, pat, case=True, flags=0, na=np.NaN):
         )
 
     def extract(self, pat, flags=0, expand=True):
+        regex = re.compile(pat, flags=flags)
+        # need an operator that can create more columns than before
+        if expand and regex.groups == 1:
+            from .dataframe import DataFrame
+
+            df = DataFrame(
+                query_compiler=self._query_compiler.str_extract(
+                    pat, flags=flags, expand=expand
+                )
+            )
+            df.columns = _get_group_names(regex)
+            return df
         return self._default_to_pandas(
             pandas.Series.str.extract, pat, flags=flags, expand=expand
         )
```

Comment:
I would rather do it on a query compiler level, it's a PandasQC implementation detail that its `str_extract` method doesn't support actual columns expanding

Reply:
This condition is also necessary in order to determine what type to return, a dataframe or a series. We can get rid of it only if we return an additional type, for example as a string, together with QC from `QC.str_exact`. Do you like this solution more than existing one?

Reply:
@dchigarev ?

Reply:
> This condition is also necessary in order to determine what type to return

Can we just do this then?
```python
def extract(...):
    qc = self._query_compiler.str_extract(...)
    if expand:
        return DataFrame(qc)
    else:
        return Series(qc)
```

Reply:
This can only be done for `expand==True` branch. On another branch, there may be several different types (DataFrame, Series, Index).

I updated the code, based on your comments, ready for review.